Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -255,15 +255,48 @@ class MessageUpdater: Worker {
         messageId: MessageId,
         completion: ((Error?) -> Void)? = nil
     ) {
+        var reaction: MessageReactionDTO?
+        let version = UUID().uuidString
+
         let endpoint: Endpoint<EmptyResponse> = .addReaction(
             type,
             score: score,
             enforceUnique: enforceUnique,
             extraData: extraData,
             messageId: messageId
         )
-        apiClient.request(endpoint: endpoint) {
-            completion?($0.error)
+
+        database.write { session in
+            do {
+                reaction = try session.addReaction(
+                    to: messageId,
+                    type: type,
+                    score: score,
+                    extraData: extraData
+                )
+                completion?(nil)
+            } catch {
+                log.warning("Failed to add the reaction to the database: \(error)")
+                completion?(error)
```

Comment:
On the other hand, this is not really true. Right now we are return `nil` on the `addReaction()` function but we should not. We should throw always an error. If this fails, the database write will also fail. So it is better to handle the DB error on the completion block

Reply:
tbh I am not sure, when this happens (writing the reaction to the DB for optimistic reasons) it still makes sense to make the API call and let that settle the whole situation